phpBB/phpbb/db/migration/data/v33x/profilefield_youtube_update.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\db\migration\data\v33x;
+
+class profilefield_youtube_update extends \phpbb\db\migration\migration
+{
+	protected $youtube_url_matcher = 'https:\\/\\/(www\\.)?youtube\\.com\\/.+';
+
+	public function effectively_installed()
+	{
+		$profile_fields = $this->table_prefix . 'profile_fields';
+
+		$result = $this->db->sql_query(
+			"SELECT field_validation
+				FROM $profile_fields
+				WHERE field_name = 'phpbb_youtube'"
+		);
+
+		$row = $this->db->sql_fetchrow($result);
+		$this->db->sql_freeresult($result);
+
+		return $row['field_validation'] === $this->youtube_url_matcher;
+	}
+
+	static public function depends_on()
+	{
+		return ['\phpbb\db\migration\data\v33x\v337'];
+	}
+
+	public function update_data()
+	{
+		return [['custom', [[$this, 'update_youtube_profile_field']]]];
+	}
+
+	public function update_youtube_profile_field()
+	{
+		$profile_fields = $this->table_prefix . 'profile_fields';
+		$profile_fields_data = $this->table_prefix . 'profile_fields_data';
+
+		$field_validation = $this->db->sql_escape($this->youtube_url_matcher);
+
+		$min_length = strlen('https://youtube.com/c/') + 1;
+
+		$this->db->sql_query(
+			"UPDATE $profile_fields SET
+				field_length = '40',
+				field_minlen = '$min_length',
+				field_maxlen = '255',
+				field_validation = '$field_validation',
+				field_contact_url = '%s'
+				WHERE field_name = 'phpbb_youtube'"
+		);
+
+		$prepend_youtube_url = $this->db->sql_concatenate(
+			"'https://youtube.com/user/'", 'pf_phpbb_youtube'
+		);
+
+		$this->db->sql_query(
+			"UPDATE $profile_fields_data SET
+				pf_phpbb_youtube = $prepend_youtube_url
+				WHERE pf_phpbb_youtube != ''
```

Comment:
Please use `<>` instead of `!=`

https://area51.phpbb.com/docs/dev/master/development/php/layout_guidelines.html#avoid-db-specific-sql

Reply:
You should be able to use `WHERE pf_phpbb_youtube IS NOT NULL` I think, instead. Not tested though.

Reply:
> You should be able to use `WHERE pf_phpbb_youtube IS NOT NULL` I think, instead. Not tested though.

I don't think so, the column is already non-nullable with a default value of empty string, so that'd be a semantic change. I'll use `<>` though, thanks for the tip!